osu.Game/Screens/Select/MatchSongSelect.cs

```diff
@@ -43,10 +52,18 @@ protected override bool OnStart()
 
         public override bool OnExiting(IScreen next)
         {
+            if (base.OnExiting(next))
+                return true;
+
+            Beatmap.Value = beatmaps.GetWorkingBeatmap(CurrentItem.Value?.Beatmap);
+            Ruleset.Value = CurrentItem.Value?.Ruleset;
+            CurrentItem.Value?.RequiredMods.Clear();
+            CurrentItem.Value?.RequiredMods.AddRange(SelectedMods.Value);
```

Comment:
If you go into song select, change mods, and press escape, this will overwrite the playlist's mods, yeah?

Not sure we want that, rather than setting the mods of the game (see `MatchSubScreen`).